CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -1145,19 +1145,9 @@ public static function processOnBehalfOrganization(&$behalfOrganization, &$conta
     // create employer relationship with $contactID only when new organization is there
     // else retain the existing relationship
     else {
-      // get the Employee relationship type id
-      $relTypeId = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_RelationshipType', 'Employee of', 'id', 'name_a_b');
-
-      // keep relationship params ready
-      $relParams['relationship_type_id'] = $relTypeId . '_a_b';
-      $relParams['is_permission_a_b'] = 1;
-      $relParams['is_active'] = 1;
       $isNotCurrentEmployer = TRUE;
     }
 
-    // formalities for creating / editing organization.
-    $behalfOrganization['contact_type'] = 'Organization';
```

Comment:
If this line is removed then when the array gets passed to hooks via createProfileContact(), the array won't contain `contact_type` anymore. Unless it's getting set somewhere else in between but I can't see where.

Reply:
Oh, you're right.  I saw that "Organization" was passed as the `$ctype` argument to `createProfileContact()`, but that doesn't get passed to the hook.  I'll restore this line and put it just above `createProfileContact()` since it's not needed by the other functions that use `$behalfOrganization` as an argument.